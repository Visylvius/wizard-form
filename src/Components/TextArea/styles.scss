@import '../../variables';

.TextArea-label { 
  color: $font-main;
  font-size: 12px;
}

.TextArea-component {
  border: 1px solid $textarea-background;
  border-radius: 3px;
  color: $font-main;
  font-size: 12px;
  outline: none;
  height: 54px;
  margin-top: 10px;
  width: 100%;
}