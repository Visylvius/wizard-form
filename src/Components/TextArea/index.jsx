import React from 'react';
import classNames from 'classnames';

import './styles.scss';

const TextArea = ({ hasLabel, label, onChange, textAreaClasses, ...rest }) => {
  const textAreaClass = classNames('TextArea-component', textAreaClasses);

  return (
    <>
      {hasLabel && <label className='TextArea-label'>{label}</label>}
      <textarea className={textAreaClass} onChange={onChange} {...rest} />
    </>
  );
};

export default TextArea;
