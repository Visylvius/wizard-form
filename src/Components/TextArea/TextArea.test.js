import React from 'react';
import renderer from 'react-test-renderer';

import TextArea from './index';

describe('render', () => {
  test('should render properly', () => {
    const tree = renderer.create(
      <TextArea hasLabel label='this is a label' value='hello' />,
    );

    expect(tree).toMatchSnapshot();
  });
});
