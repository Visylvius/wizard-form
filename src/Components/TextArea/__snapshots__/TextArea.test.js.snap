// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render properly 1`] = `
Array [
  <label
    className="TextArea-label"
  >
    this is a label
  </label>,
  <textarea
    className="TextArea-component"
    value="hello"
  />,
]
`;
